array_family/flex_ext.py

```diff
@@ -606,6 +591,72 @@ def __init__(self):
         mask2.set_selected(sind.select(mask), True)
         return mask2, other_matched, other_unmatched
 
+    def match(
+        self,
+        other: dials_array_family_flex_ext.reflection_table,
+        max_separation: int = 2,
```

Comment:
This might make sense either `key`-first, or keyword-only. `rt.match(other, "xyz")` looks right but isn't
```suggestion
        *,
        max_separation: int = 2,
```